Snowflake.Data.Tests/UnitTests/ConnectionPoolManagerTest.cs

```diff
@@ -0,0 +1,323 @@
+/*
+ * Copyright (c) 2023 Snowflake Computing Inc. All rights reserved.
+ */
+
+using System.Collections.Generic;
+using System.Security;
+using System.Threading;
+using System.Threading.Tasks;
+using NUnit.Framework;
+using Snowflake.Data.Core;
+using Snowflake.Data.Core.Session;
+using Moq;
+using Snowflake.Data.Client;
+using Snowflake.Data.Tests.Util;
+
+namespace Snowflake.Data.Tests.UnitTests
+{
+    [TestFixture, NonParallelizable]
+    class ConnectionPoolManagerTest
+    {
+        private readonly ConnectionPoolManager _connectionPoolManager = new ConnectionPoolManager();
+        private const string ConnectionString1 = "database=D1;warehouse=W1;account=A1;user=U1;password=P1;role=R1;";
+        private const string ConnectionString2 = "database=D2;warehouse=W2;account=A2;user=U2;password=P2;role=R2;";
+        private readonly SecureString _password = new SecureString();
+        private static PoolConfig s_poolConfig;
+
+        [OneTimeSetUp] 
+        public static void BeforeAllTests()
+        {
+            s_poolConfig = new PoolConfig();
+            SnowflakeDbConnectionPool.SetConnectionPoolVersion(ConnectionPoolType.MultipleConnectionPool);
+            SessionPool.sessionFactory = new MockSessionFactory();
+        }
+        
+        [OneTimeTearDown]
+        public void AfterAllTests()
+        {
+            s_poolConfig.Reset();
+            SessionPool.sessionFactory = new SessionFactory();
+        }
+
+        [Test]
+        public void TestPoolManagerReturnsSessionPoolForGivenConnectionString()
+        {
+            // Act
+            var sessionPool = _connectionPoolManager.GetPool(ConnectionString1, _password);
+            
+            // Assert
+            Assert.AreEqual(ConnectionString1, sessionPool._connectionString);
```

Comment:
I feel a little bit uncomfortable seeing getting a private field: 
```
sessionPool._connectionString
```

maybe for these fileds (connectionString and password) could be created internal properties so here we could reference internal property instead of private field?

Reply:
They are internal

Reply:
maybe make it internal property ConnectionString with get only?